mne/io/tests/test_base.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+"""Run tests for the BaseRaw class."""
+# Author: Stefan Appelhoff <stefan.appelhoff@mailbox.org>
+#
+# License: BSD (3-clause)
+
+import pytest
+
+from mne import create_info
+from mne.io import BaseRaw
+
+
+def test_orig_units():
+    """Test the error handling for original units."""
+    # Should work fine
+    info = create_info(ch_names=['Cz'], sfreq=100, ch_types='eeg')
+    BaseRaw(info, last_samps=[1], orig_units={'Cz': 'nV'})
+
+    # Should complain that channel Cz does not have a corresponding original
+    # unit.
+    with pytest.raises(ValueError, match='has no associated original unit.'):
+        info = create_info(ch_names=['Cz'], sfreq=100, ch_types='eeg')
+        BaseRaw(info, last_samps=[1], orig_units={'not_Cz': 'nV'})
+
+    # Test that a non-dict orig_units argument raises a ValueError
+    with pytest.raises(ValueError, match='orig_units must be of type dict'):
+        info = create_info(ch_names=['Cz'], sfreq=100, ch_types='eeg')
+        BaseRaw(info, last_samps=[1], orig_units=True)
```

Comment:
on a second though I think this should be put in this file:

https://github.com/mne-tools/mne-python/blob/f81bf9b96c7e530148e8781357d9e8a71fc9a418/mne/io/tests/test_raw.py

and I would update the common raw test:

https://github.com/mne-tools/mne-python/blob/f81bf9b96c7e530148e8781357d9e8a71fc9a418/mne/io/tests/test_raw.py#L24:5

so you have a test that orig_units is present for all raw readers.

Reply:
so you suggest that I migrate this code from `test_base.py` into the function `_test_raw_reader` within `test_raw.py`?

Or that I create a new test function within `test_raw.py`?

Reply:
@agramfort @larsoner @massich I don't really understand what you are planning. The tests here in `test_base.py` 
should be moved somewhere else?

1. Why? As far as I see it the tests here make sure that functionality in base.py is working properly, so it seemed like a logical place. I would keep it here and simply add a line `assert hasattr(raw, '_orig_units')` in `_test_raw_reader`

2. Where - you say to `test_raw.py`?

3. How - in a new `test_orig_units` function within `test_raw.py` ... or as part of `_test_raw_reader` within `test_raw.py`

Reply:
Looking closer I disagree @agramfort -- this test looks at corner cases of `BaseRaw.__init__`, so this is a reasonable place (and should only be run once!). @sappelhoff already has tests that `_orig_units` is present and is `None` or `dict` below in `_test_raw_reader` so all readers are tested.

Eventually we should be able to get rid of the `if raw._orig_units is not None:` conditional, but not until we extend it to all readers

Reply:
... ahh @sappelhoff I think maybe all you need to do (since you already have the `_test_raw_reader` tests, @agramfort probably just missed them) is move/copy-paste this test function to `mne/io/tests/test_raw.py` rather than creating this new file. Sorry I led you astray on that!